src/main/recorder.ts

```diff
@@ -482,6 +499,42 @@ class Recorder {
 
     console.debug('[OBS] Asserted OBS signal:', value);
   };
+
+  static getInputAudioDevices() {
+    // @@@ todo check init or fail
+    const inputDevices: IDevice[] = [];
+
+    (osn.NodeObs.OBS_settings_getInputAudioDevices() as IOBSDevice[]).forEach(
```

Comment:
Would look slightly cleaner as an assignment of `const inputDevices =  (osn.NodeObs.OBS_settings_getInputAudioDevices() as IOBSDevice[]).map(...)` I think as that'd make it immediately obvious what's going on.

Reply:
Agree this was super ugly - I think I've gone one step further and removed all the map stuff entirely as it was to add a parameter that was unused! This code was copy pasted straight out of the streamlabs desktop repository... much happier with it now as:

```
  getInputAudioDevices() {
    console.info('[Recorder] Getting available input devices');

    if (!this.obsInitialized) {
      throw new Error('[Recorder] OBS not initialized');
    }

    const inputDevices =
      osn.NodeObs.OBS_settings_getInputAudioDevices() as IOBSDevice[];

    return inputDevices.filter((v) => v.id !== 'default');
  }
```